frameworks/js-bindings/bindings/manual/network/XMLHTTPRequest.cpp

```diff
@@ -724,10 +731,28 @@ JS_BINDED_FUNC_IMPL(MinXmlHttpRequest, send)
     _setHttpRequestHeader();
     _sendRequest(cx);
     _notify(_onloadstartCallback);
+    
+    //begin schedule for timeout
+    if(_timeout > 0)
+    {
+        _scheduler->scheduleUpdate(this, 0, false);
```

Comment:
@joshuastray, 
> are you sure the xhr timeout means timeout for connect in the w3c specification?

I don't know, but HttpClient supports both of them, including connection timeout & read timeout.

> when you notify the callback, it's the right time when time's out? not some time delayed?
You could look at the implementation of `HttpClient`, if timeout happens, it will notify the callback in GL thread by using `performFunctionInCocosThread`. So it's true there will be little time delayed.